### PR TITLE
[JBPM-9310] Warning in BAMTaskEventListener if task is skipped from t…

### PR DESCRIPTION
**Pull request by elguardian**
Jira:
https://issues.redhat.com/browse/JBPM-9310

### CODE REVIEW

**Comment by elguardian** [comment]
depends on 
https://github.com/kiegroup/droolsjbpm-knowledge/pull/460
https://github.com/kiegroup/jbpm/pull/1741
https://github.com/kiegroup/droolsjbpm-integration/pull/2211

**Comment by elguardian** [comment]
Jenkins retest this please.

**Comment by afalhambra** [comment]
Jenkins run fdb

File: kie-internal/src/main/java/org/kie/internal/runtime/manager/deploy/DeploymentDescriptorImpl.java [hunk 1]
```diff
@@ -340,7 +341,13 @@ public void setLimitSerializationClasses(Boolean limitSerializationClasses) {
         return input;
     }
 
-    protected void removeTransient(Set<?> input) {
+    protected List<ObjectModel> cleanList(List<ObjectModel> input) {
+        input.remove(null);
```
Comment (gmunozfe) [inline kie-internal/src/main/java/org/kie/internal/runtime/manager/deploy/DeploymentDescriptorImpl.java:345]: For removing all nulls (if more than one), it should be a while loop:
```suggestion
        while (input.remove(null));
```

**Review by gmunozfe — COMMENTED** [review]
Looks good, just a single comment

File: kie-internal/src/main/java/org/kie/internal/runtime/manager/deploy/DeploymentDescriptorImpl.java [hunk 2]
```diff
@@ -82,7 +83,7 @@
 
     @XmlElement(name = "task-event-listener")
     @XmlElementWrapper(name = "task-event-listeners")
-    private Set<ObjectModel> taskEventListeners = new LinkedHashSet<ObjectModel>();
+    private List<ObjectModel> taskEventListeners = new ArrayList<ObjectModel>();
```
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-internal/src/main/java/org/kie/internal/runtime/manager/deploy/DeploymentDescriptorImpl.java:86]: done

File: kie-internal/src/main/java/org/kie/internal/runtime/manager/deploy/DeploymentDescriptorImpl.java [hunk 3]
```diff
@@ -280,7 +281,7 @@ public void setEventListeners(List<ObjectModel> eventListeners) {
 
     public void setTaskEventListeners(List<ObjectModel> taskEventListeners) {
         if (taskEventListeners != null) {
-            this.taskEventListeners = new HashSet<ObjectModel>(taskEventListeners);
+            this.taskEventListeners = new ArrayList<ObjectModel>(taskEventListeners);
```
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-internal/src/main/java/org/kie/internal/runtime/manager/deploy/DeploymentDescriptorImpl.java:284]: done

File: kie-internal/src/main/java/org/kie/internal/runtime/manager/deploy/DeploymentDescriptorImpl.java [hunk 4]
```diff
@@ -503,7 +510,7 @@ public DeploymentDescriptorBuilder addWorkItemHandler(NamedObjectModel model) {
             @Override
             public DeploymentDescriptorBuilder addTaskEventListener(ObjectModel model) {
                 if (handler.accepted(model)) {
-                    if (!descriptor.taskEventListeners.add(model)) {
+                    if (!descriptor.taskEventListeners.contains(model)) {
                         descriptor.taskEventListeners.remove(model);
                         descriptor.taskEventListeners.add(model);
```
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-internal/src/main/java/org/kie/internal/runtime/manager/deploy/DeploymentDescriptorImpl.java:515]: done

File: kie-internal/src/main/java/org/kie/internal/runtime/manager/deploy/DeploymentDescriptorImpl.java (hunk 1 above)
Comment (elguardian) [reply kie-internal/src/main/java/org/kie/internal/runtime/manager/deploy/DeploymentDescriptorImpl.java]: done

**Review by gmunozfe — APPROVED** [review]
Looks good to me, nailed down @elguardian !

**Review by afalhambra — APPROVED** [review]
Very good job @elguardian

**Pull request merged**